### ISSUE TITLE
Typo in a Test2::API::Breakage warning

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#17137]
Migrated from [rt.perl.org#134381](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=134381) (status was 'pending release')

Searchable as RT134381$

**Comment by p5pRT** [comment]
### From choroba@matfyz.cz

#### Created by choroba@matfyz.cz



Test2 "recommeds" to update related test modules instead of
"recommending" to update them\. See the attached patch\.

Ch\.

<details><summary>Perl Info</summary>

```

Flags:
     category=library
     severity=low

Site configuration information for perl 5.31.4:

Configured by choroba at Mon Aug 26 16:15:05 CEST 2019.

Summary of my perl5 (revision 5 version 31 subversion 4) configuration:
   Commit id: 6e404ab585deadc1c32d50513f13b50ae395c00d
   Platform:
     osname=linux
     osvers=4.12.14-lp151.28.13-default
     archname=x86_64-linux-thread-multi
     uname='linux lenonovo 4.12.14-lp151.28.13-default #1 smp wed aug 7 07:20:16 utc 2019 (0c09ad2) x86_64 x86_64 x86_64 gnulinux '
     config_args='-rdes -Dusethreads -Dpthread -Dprefix=~/blead -Dusedevel'
     hint=recommended
     useposix=true
     d_sigaction=define
     useithreads=define
     usemultiplicity=define
     use64bitint=define
     use64bitall=define
     uselongdouble=undef
     usemymalloc=n
     default_inc_excludes_dot=define
     bincompat5005=undef
   Compiler:
     cc='cc'
     ccflags ='-D_REENTRANT -D_GNU_SOURCE -fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -D_FORTIFY_SOURCE=2'
     optimize='-O2'
     cppflags='-D_REENTRANT -D_GNU_SOURCE -fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include'
     ccversion=''
     gccversion='7.4.1 20190424 [gcc-7-branch revision 270538]'
     gccosandvers=''
     intsize=4
     longsize=8
     ptrsize=8
     doublesize=8
     byteorder=12345678
     doublekind=3
     d_longlong=define
     longlongsize=8
     d_longdbl=define
     longdblsize=16
     longdblkind=3
     ivtype='long'
     ivsize=8
     nvtype='double'
     nvsize=8
     Off_t='off_t'
     lseeksize=8
     alignbytes=8
     prototype=define
   Linker and Libraries:
     ld='cc'
     ldflags =' -fstack-protector-strong -L/usr/local/lib'
     libpth=/usr/local/lib /usr/lib64/gcc/x86_64-suse-linux/7/include-fixed /usr/lib64/gcc/x86_64-suse-linux/7/../../../../x86_64-suse-linux/lib /usr/lib /lib/../lib64 /usr/lib/../lib64 /lib /lib64 /usr/lib64 /usr/local/lib64
     libs=-lpthread -lgdbm -ldl -lm -lcrypt -lutil -lc -lgdbm_compat
     perllibs=-lpthread -ldl -lm -lcrypt -lutil -lc
     libc=libc-2.26.so
     so=so
     useshrplib=false
     libperl=libperl.a
     gnulibc_version='2.26'
   Dynamic Linking:
     dlsrc=dl_dlopen.xs
     dlext=so
     d_dlsymun=undef
     ccdlflags='-Wl,-E'
     cccdlflags='-fPIC'
     lddlflags='-shared -O2 -L/usr/local/lib -fstack-protector-strong'



@INC for perl 5.31.4:
     /home/choroba/blead/lib/perl5/site_perl/5.31.4/x86_64-linux-thread-multi
     /home/choroba/blead/lib/perl5/site_perl/5.31.4
     /home/choroba/blead/lib/perl5/5.31.4/x86_64-linux-thread-multi
     /home/choroba/blead/lib/perl5/5.31.4


Environment for perl 5.31.4:
     HOME=/home/choroba
     LANG=en_US.utf8
     LANGUAGE (unset)
     LC_CTYPE=en_US.UTF-8
     LD_LIBRARY_PATH (unset)
     LOGDIR (unset)
     PATH=/home/choroba/bin:/usr/local/bin:/usr/bin:/bin:/usr/bin/X11:/home/choroba/perl5/bin:/home/choroba/opensource/worktime/bin:.
     PERL_BADLANG (unset)
     SHELL=/bin/bash
```
</details>

**Comment by p5pRT** [comment]
### From choroba@matfyz.cz

<details><summary>0001-Fix-a-typo-in-a-Test2-API-Breakage-warning.patch</summary>

```diff
From 8682e2afe6daee5169de8b1e68c85ebbab3635a2 Mon Sep 17 00:00:00 2001
From: "E.Choroba" <choroba@matfyz.cz>
Date: Mon, 26 Aug 2019 17:42:48 +0200
Subject: [PATCH] Fix a typo in a Test2::API::Breakage warning

---
 cpan/Test-Simple/lib/Test2/API/Breakage.pm      | 2 +-
 cpan/Test-Simple/t/Test2/modules/API/Breakage.t | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/cpan/Test-Simple/lib/Test2/API/Breakage.pm b/cpan/Test-Simple/lib/Test2/API/Breakage.pm
index 106748a513..77b97c31ac 100644
--- a/cpan/Test-Simple/lib/Test2/API/Breakage.pm
+++ b/cpan/Test-Simple/lib/Test2/API/Breakage.pm
@@ -73,7 +73,7 @@ sub report {
         next unless $INC{$file} || ($require && eval { require $file; 1 });
         my $want = $suggest{$mod};
         next if eval { $mod->VERSION($want); 1 };
-        push @warn => " * Module '$mod' is outdated, we recommed updating above $want.";
+        push @warn => " * Module '$mod' is outdated, we recommend updating above $want.";
     }
 
     for my $mod (keys %required) {
diff --git a/cpan/Test-Simple/t/Test2/modules/API/Breakage.t b/cpan/Test-Simple/t/Test2/modules/API/Breakage.t
index e2932469f3..f5af6c59a6 100644
--- a/cpan/Test-Simple/t/Test2/modules/API/Breakage.t
+++ b/cpan/Test-Simple/t/Test2/modules/API/Breakage.t
@@ -53,7 +53,7 @@ for my $meth (qw/upgrade_suggested upgrade_required known_broken/) {
             [sort @report],
             [
                 sort
-                " * Module 'T2Test::UG1' is outdated, we recommed updating above 1.0.",
+                " * Module 'T2Test::UG1' is outdated, we recommend updating above 1.0.",
                 " * Module 'T2Test::UR1' is outdated and known to be broken, please update to 1.0 or higher.",
                 " * Module 'T2Test::KB1' is known to be broken in version 1.0 and below, newer versions have not been tested. You have: 0.9",
                 " * Module 'T2Test::KB2' is known to be broken in version 0.5 and below, newer versions have not been tested. You have: 0.9",
-- 
2.16.4


```
</details>

**Comment by p5pRT** [comment]
### From @grinnz

On Mon\, 26 Aug 2019 08&#8203;:45&#8203;:22 \-0700\, choroba@&#8203;matfyz\.cz wrote&#8203;:
> This is a bug report for perl from choroba@&#8203;matfyz\.cz\,
> generated with the help of perlbug 1\.41 running under perl 5\.31\.4\.
> 
> 
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> \[Please describe your issue here\]
> 
> Test2 "recommeds" to update related test modules instead of
> "recommending" to update them\. See the attached patch\.
> 
> Ch\.
> 
> 
> \[Please do not change anything below this line\]
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> \-\-\-
> Flags&#8203;:
>      category=library
>      severity=low
> \-\-\-
> Site configuration information for perl 5\.31\.4&#8203;:
> 
> Configured by choroba at Mon Aug 26 16&#8203;:15&#8203;:05 CEST 2019\.
> 
> Summary of my perl5 \(revision 5 version 31 subversion 4\)
> configuration&#8203;:
>    Commit id&#8203;: 6e404ab585deadc1c32d50513f13b50ae395c00d
>    Platform&#8203;:
>      osname=linux
>      osvers=4\.12\.14\-lp151\.28\.13\-default
>      archname=x86\_64\-linux\-thread\-multi
>      uname='linux lenonovo 4\.12\.14\-lp151\.28\.13\-default \#1 smp wed aug
> 7 07&#8203;:20&#8203;:16 utc 2019 \(0c09ad2\) x86\_64 x86\_64 x86\_64 gnulinux '
>      config\_args='\-rdes \-Dusethreads \-Dpthread \-Dprefix=~/blead
> \-Dusedevel'
>      hint=recommended
>      useposix=true
>      d\_sigaction=define
>      useithreads=define
>      usemultiplicity=define
>      use64bitint=define
>      use64bitall=define
>      uselongdouble=undef
>      usemymalloc=n
>      default\_inc\_excludes\_dot=define
>      bincompat5005=undef
>    Compiler&#8203;:
>      cc='cc'
>      ccflags ='\-D\_REENTRANT \-D\_GNU\_SOURCE \-fwrapv \-fno\-strict\-aliasing
> \-pipe \-fstack\-protector\-strong \-I/usr/local/include
> \-D\_LARGEFILE\_SOURCE \-D\_FILE\_OFFSET\_BITS=64 \-D\_FORTIFY\_SOURCE=2'
>      optimize='\-O2'
>      cppflags='\-D\_REENTRANT \-D\_GNU\_SOURCE \-fwrapv \-fno\-strict\-aliasing
> \-pipe \-fstack\-protector\-strong \-I/usr/local/include'
>      ccversion=''
>      gccversion='7\.4\.1 20190424 \[gcc\-7\-branch revision 270538\]'
>      gccosandvers=''
>      intsize=4
>      longsize=8
>      ptrsize=8
>      doublesize=8
>      byteorder=12345678
>      doublekind=3
>      d\_longlong=define
>      longlongsize=8
>      d\_longdbl=define
>      longdblsize=16
>      longdblkind=3
>      ivtype='long'
>      ivsize=8
>      nvtype='double'
>      nvsize=8
>      Off\_t='off\_t'
>      lseeksize=8
>      alignbytes=8
>      prototype=define
>    Linker and Libraries&#8203;:
>      ld='cc'
>      ldflags =' \-fstack\-protector\-strong \-L/usr/local/lib'
>      libpth=/usr/local/lib /usr/lib64/gcc/x86\_64\-suse\-linux/7/include\-
> fixed /usr/lib64/gcc/x86\_64\-suse\-linux/7/\.\./\.\./\.\./\.\./x86\_64\-suse\-
> linux/lib /usr/lib /lib/\.\./lib64 /usr/lib/\.\./lib64 /lib /lib64
> /usr/lib64 /usr/local/lib64
>      libs=\-lpthread \-lgdbm \-ldl \-lm \-lcrypt \-lutil \-lc \-lgdbm\_compat
>      perllibs=\-lpthread \-ldl \-lm \-lcrypt \-lutil \-lc
>      libc=libc\-2\.26\.so
>      so=so
>      useshrplib=false
>      libperl=libperl\.a
>      gnulibc\_version='2\.26'
>    Dynamic Linking&#8203;:
>      dlsrc=dl\_dlopen\.xs
>      dlext=so
>      d\_dlsymun=undef
>      ccdlflags='\-Wl\,\-E'
>      cccdlflags='\-fPIC'
>      lddlflags='\-shared \-O2 \-L/usr/local/lib \-fstack\-protector\-strong'
> 
> 
> \-\-\-
> @&#8203;INC for perl 5\.31\.4&#8203;:
>      /home/choroba/blead/lib/perl5/site\_perl/5\.31\.4/x86\_64\-linux\-
> thread\-multi
>      /home/choroba/blead/lib/perl5/site\_perl/5\.31\.4
>      /home/choroba/blead/lib/perl5/5\.31\.4/x86\_64\-linux\-thread\-multi
>      /home/choroba/blead/lib/perl5/5\.31\.4
> 
> \-\-\-
> Environment for perl 5\.31\.4&#8203;:
>      HOME=/home/choroba
>      LANG=en\_US\.utf8
>      LANGUAGE \(unset\)
>      LC\_CTYPE=en\_US\.UTF\-8
>      LD\_LIBRARY\_PATH \(unset\)
>      LOGDIR \(unset\)
>      PATH=/home/choroba/bin&#8203;:/usr/local/bin&#8203;:/usr/bin&#8203;:/bin&#8203;:/usr/bin/X11&#8203;:/home/choroba/perl5/bin&#8203;:/home/choroba/opensource/worktime/bin&#8203;:\.
>      PERL\_BADLANG \(unset\)
>      SHELL=/bin/bash

FYI\, Test2 is maintained on CPAN&#8203;: https://github.com/Test-More/test-more

\-Dan

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From @exodist

Patch applied to the cpan module source code&#8203;: https://github.com/Test-More/test-more/commit/9b595a698bb4b83febb4421be7edd665a926a30a

Next time I mint a release this will go out\, and we can then close this ticket\.

**Comment by p5pRT** [comment]
### From @jkeenan

On Mon\, 26 Aug 2019 21&#8203;:09&#8203;:15 GMT\, exodist7@&#8203;gmail\.com wrote&#8203;:
> Patch applied to the cpan module source code&#8203;: https://github.com/Test-
> More/test\-more/commit/9b595a698bb4b83febb4421be7edd665a926a30a
> 
> Next time I mint a release this will go out\, and we can then close
> this ticket\.

This was fixed in commit 00ea29f1a18 \(Chad Granum             2019\-09\-13\)\.

Thank you very much\.
\-\- 
James E Keenan \(jkeenan@&#8203;cpan\.org\)

**Comment by p5pRT** [comment]
@jkeenan - Status changed from 'open' to 'pending release'
